app/src/main/kotlin/at/bitfire/davdroid/ui/DebugInfoActivity.kt

```diff
@@ -120,101 +166,216 @@ class DebugInfoActivity : AppCompatActivity() {
+                    }
+                }
+            }
+            modelCause?.let { cause ->
+                item {
+                    CardWithImage(
+                        image = painterResource(R.drawable.undraw_server_down),
+                        title = when (cause) {
+                            is HttpException -> stringResource(if (cause.code / 100 == 5) R.string.debug_info_server_error else R.string.debug_info_http_error)
+                            is DavException -> stringResource(R.string.debug_info_webdav_error)
+                            is IOException, is IOError -> stringResource(R.string.debug_info_io_error)
+                            else -> cause::class.java.simpleName
+                        },
+                        subtitle = cause.localizedMessage,
+                        message = stringResource(
+                            if (cause is HttpException)
+                                when {
+                                    cause.code == 403 -> R.string.debug_info_http_403_description
+                                    cause.code == 404 -> R.string.debug_info_http_404_description
+                                    cause.code / 100 == 5 -> R.string.debug_info_http_5xx_description
+                                    else -> R.string.debug_info_unexpected_error
+                                }
+                            else
+                                R.string.debug_info_unexpected_error
+                        ),
+                        icon = Icons.Rounded.Info,
+                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp)
+                    ) {
+                        TextButton(
+                            enabled = debugInfo != null,
+                            onClick = { shareFile(debugInfo!!) }
+                        ) {
+                            Text(stringResource(R.string.debug_info_view_details))
+                        }
+                    }
+                }
+            }
+            debugInfo?.let { info ->
+                item {
+                    CardWithImage(
+                        image = painterResource(R.drawable.undraw_server_down),
+                        title = stringResource(R.string.debug_info_title),
+                        subtitle = stringResource(R.string.debug_info_subtitle),
+                        icon = Icons.Rounded.BugReport,
+                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp)
+                    ) {
+                        TextButton(
+                            onClick = { shareFile(info) }
+                        ) {
+                            Text(stringResource(R.string.debug_info_view_details))
+                        }
+                    }
+                }
+            }
+            (localResource to remoteResource)
+                .takeIf { (l, r) -> l != null || r != null }
+                ?.let { (local, remote) ->
```

Comment:
This expression is too complex to understand it at a first glance, which reduces maintainability. Maybe that one instead?

```
val localResourceConst = localResource
val remoteResourceConst = remoteResource
if (localResourceConst != null || remoteResourceConst != null) {
  // work with localResourceConst and remoteResourceConst
}
```